processor/processor.go

```diff
@@ -903,7 +903,7 @@ func (proc *HandleT) mainLoop() {
 	currLoopSleep := time.Duration(0)
 
 	for {
-		if proc.handlePendingGatewayJobs() {
+		if !proc.handlePendingGatewayJobs() {
```

Comment:
This is the main reason this PR was raised. Needs a careful review.

Reply:
Looks good. 

My only comment is I don't prefer negative if conditions. Makes it hard to read.

Reply:
Instead of the whole logic below, how about using https://github.com/cenkalti/backoff/